### ISSUE TITLE
Applying any XSLT will change the original XML indentation

### ISSUE THREAD

**Issue by streino** [ecolabdata/ecospheres-isomorphe#83]
For instance, the following attribute indentation : 
```
<gmd:role>
  <gmd:CI_RoleCode codeList="http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/resources/codelist/ML_gmxCodelists.xml#CI_RoleCode"
                   codeListValue="pointOfContact"/>
</gmd:role>
```
becomes : 
```
<gmd:role>
  <gmd:CI_RoleCode codeList="http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/resources/codelist/ML_gmxCodelists.xml#CI_RoleCode" codeListValue="pointOfContact"/>
</gmd:role>
```

Logically it doesn't matter, but this introduce (transparent) changes in the original catalog.

Do we want to leave any XSLT silently reindent, or should we require a first explicit pass to apply standard indentation, before any real transformation occurs?

Pros of staying silent : 
- Don't apply needless reindenting to records that won't be affected by other transformations.
- Easiest technical solution (= do nothing).

Pros of explicit pass : 
- Transparent to the user.
- Less noise in diffs (but we could indent both versions before doing diffs to mitigate that).

Cons of any solution I could think of so far : 
- Conflict with Geonetwork indenting, so there will be a back and forth when we apply a transformation, and when Geonetwork does. I don't know how they manage to maintain what seems like custom indent when they apply batch transforms on their side...

**Comment by streino** [comment]
Geonetwork uses JDOM "pretty format" : e.g. https://github.com/geonetwork/core-geonetwork/blob/main/common/src/main/java/org/fao/geonet/utils/Xml.java#L587

I can't find a similar formatter for python.

**Comment by streino** [comment]
Turns out it doesn't matter, as Geonetwork will reformat any update we submit.

The only "issue" is if someone compares the original XML from Geonetwork to the transformed XML from ISOmorphe. Like I did when investigating other tickets... As long as we remember to only compare original/transformed within one system (either ISOmorphe or Geonetwork), we'll be fine. And if comparing across systems, remember to reformat both XMLs with something like noop.xsl before trying to diff.
